catalog/core/catalog-core-versioning/versioning-common/src/main/java/ddf/catalog/core/versioning/MetacardVersion.java

```diff
@@ -209,31 +260,83 @@ public MetacardVersion(Metacard sourceMetacard, Action action, Subject subject)
         this.setTags(Collections.singleton(VERSION_TAG));
     }
 
+    private byte[] getVersionType(Metacard sourceMetacard) {
+        try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                ObjectOutputStream oos = new ObjectOutputStream(baos)) {
+            oos.writeObject(sourceMetacard.getMetacardType());
```

Comment:
You need to put the data from the metacard type into an implementation that this bundle will always have access to which would most likely be the MetacardTypeImpl that is embedded in the bundle.  Otherwise this class could throw class not found exceptions if other bundles use custom, embedded metacard type implementations.